src/applicationcomposer/messageHandlers/deployMessageHandler.ts

```diff
@@ -3,10 +3,49 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import vscode from 'vscode'
+import vscode, { Uri } from 'vscode'
 import { Command, DeployRequestMessage, DeployResponseMessage, MessageType, WebviewContext } from '../types'
 import { SamSyncResult } from '../../shared/sam/sync'
 import { telemetry } from '../../shared/telemetry/telemetry'
+import { Auth } from '../../auth/auth'
+import { promptAndUseConnection } from '../../auth/utils'
+import { openUrl } from '../../shared/utilities/vsCodeUtils'
+import { localize } from 'vscode-nls'
+import { StatefulConnection } from '../../auth/connection'
+
+const credentialsDocLink = 'https://docs.aws.amazon.com/toolkit-for-vscode/latest/userguide/setup-credentials.html'
+
+async function promptReauth(connection?: StatefulConnection) {
+    let errorMessage = localize(
+        'aws.applicationComposer.deploy.authModal.message',
+        'Syncing requires authentication with IAM credentials.'
+    )
+    if (connection?.state === 'valid') {
+        errorMessage =
+            localize(
+                'aws.applicationComposer.deploy.authModal.invalidAuth',
+                'Authentication through Builder ID or IAM Identity Center detected. '
+            ) + errorMessage
+    }
+    const acceptMessage = localize(
+        'aws.applicationComposer.deploy.authModal.accept',
+        'Authenticate with IAM credentials'
+    )
+    const docMessage = localize('aws.applicationComposer.deploy.authModal.docLink', 'Open documentation')
+    const modalResponse = await vscode.window.showInformationMessage(
+        errorMessage,
+        { modal: true },
+        acceptMessage,
+        docMessage
+    )
+    if (modalResponse === docMessage) {
+        await openUrl(Uri.parse(credentialsDocLink))
+    }
```

Comment:
use `showMessageWithUrl` https://github.com/aws/aws-toolkit-vscode/blob/2a1b31cf3218c3eea6fa56d69a0fbd7a894eaad1/src/shared/utilities/messages.ts#L62

Reply:
This makes sense here, just updated.